contracts/javascore/aggregator/src/main/java/relay/aggregator/RelayAggregator.java

```diff
@@ -0,0 +1,305 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package relay.aggregator;
+
+import java.math.BigInteger;
+
+import score.Context;
+import score.Address;
+import score.ArrayDB;
+import score.VarDB;
+import score.DictDB;
+import score.BranchDB;
+import score.ByteArrayObjectWriter;
+import score.annotation.EventLog;
+import score.annotation.External;
+import score.ObjectReader;
+import scorex.util.ArrayList;
+import scorex.util.HashMap;
+
+public class RelayAggregator {
+    private final Integer DEFAULT_SIGNATURE_THRESHOLD = 2;
+
+    private final VarDB<Integer> signatureThreshold = Context.newVarDB("signatureThreshold", Integer.class);
+
+    private final VarDB<Address> admin = Context.newVarDB("admin", Address.class);
+
+    private final ArrayDB<Address> relayers = Context.newArrayDB("relayers", Address.class);
+
+    private final DictDB<String, Packet> packets = Context.newDictDB("packets", Packet.class);
+
+    private final BranchDB<String, DictDB<Address, byte[]>> signatures = Context.newBranchDB("signatures",
+            byte[].class);
+
+    public RelayAggregator(Address _admin) {
+        if (admin.get() == null) {
+            admin.set(_admin);
+            signatureThreshold.set(DEFAULT_SIGNATURE_THRESHOLD);
+        }
+    }
+
+    @External
+    public void setAdmin(Address _admin) {
+        adminOnly();
+        admin.set(_admin);
+    }
+
+    @External(readonly = true)
+    public Address getAdmin() {
+        return admin.get();
+    }
+
+    @External
+    public void setSignatureThreshold(int threshold) {
+        adminOnly();
+        signatureThreshold.set(threshold);
+    }
+
+    @External(readonly = true)
+    public int getSignatureThreshold() {
+        return signatureThreshold.get();
+    }
+
+    @External(readonly = true)
+    public Address[] getRelayers() {
+        Address[] rlrs = new Address[relayers.size()];
+        for (int i = 0; i < relayers.size(); i++) {
+            rlrs[i] = relayers.get(i);
+        }
+        return rlrs;
+    }
+
+    @External
+    public void addRelayers(Address[] newRelayers) {
+        adminOnly();
+
+        Context.require(newRelayers != null && newRelayers.length != 0, "new relayers cannot be empty");
+
+        HashMap<Address, Boolean> existingRelayers = new HashMap<Address, Boolean>();
+        for (int i = 0; i < relayers.size(); i++) {
+            Address relayer = relayers.get(i);
+            existingRelayers.put(relayer, true);
+        }
+
+        for (Address newRelayer : newRelayers) {
+            if (!existingRelayers.containsKey(newRelayer)) {
+                relayers.add(newRelayer);
+                existingRelayers.put(newRelayer, true);
+            }
+        }
+    }
+
+    @External
+    public void removeRelayers(Address[] relayersToBeRemoved) {
+        adminOnly();
+
+        Context.require(relayersToBeRemoved != null && relayersToBeRemoved.length != 0,
```

Comment:
just use dictdb of relayer address and boolean . would save lot of loops

Reply:
done...